### DISCUSSION TITLE
`readBinaryFile` immensely slow

### DISCUSSION THREAD

**Discussion by thaumictom** [tauri-apps/tauri#2487] [Q&A]
The issue in #1817 does not seem to be resolved yet or I somehow couldn't get the same results. I have made some testing and it just isn't fast compared to native browser functions.

I have made a new project using `create-tauri-app` with the react framework and have made a basic `readBinaryFile` function that reads a 2MB file on my PC. During debug, it took 5 seconds to return the promise, in release it took around 500 ms. It is nothing compared to the aforementioned issue stating 40 ms for 3 MB. In the browser, with a file input it takes less than 10 ms using `readAsArrayBuffer`.

**Comment by nothingismagick** [comment]
We are actually working on this right now, especially because we discovered unintended memory bloat and aren't doing streaming. Perhaps @lemarier can add more insight.

**Reply by hut36** [reply]
> ... unintended memory bloat ...

This problem still exists in tauri 2 alpha. Is there an issue to track this?

**Comment by FabianLars** [comment]
btw, just to clear things up. The issue doesn't say that it only takes 40ms.
It says it took 40ms to read the file on the rust side, but that means the file still has to be transported into the webview (to the js).

He said that in total it took 400ms in release mode for the file to arrive on the js side (40ms reading, 110ms serializing, rest is transport i guess).
The 400ms for a 3MB file should be relatively the same as your 500ms for 2MB, considering hardware differences and such.

(btw i measured ~390ms for 2.3MB with readBinaryFile and ~90ms with readTextFile.)
